src/meta/src/stream/scale.rs

```diff
@@ -549,39 +728,46 @@ where
                 let mut new_actor = sample_actor.clone();
 
                 let worker = ctx.parallel_unit_id_to_worker(new_parallel_unit_id)?;
-                for upstream_actor_id in &new_actor.upstream_actor_id {
-                    let upstream_worker_id = ctx
-                        .actor_id_to_parallel_unit(upstream_actor_id)?
-                        .worker_node_id;
-                    worker_hanging_channels
-                        .entry(upstream_worker_id)
-                        .or_default()
-                        .push(HangingChannel {
-                            upstream: Some(ActorInfo {
-                                actor_id: *upstream_actor_id,
-                                host: None,
-                            }),
-                            downstream: Some(ActorInfo {
-                                actor_id: *new_actor_id,
-                                host: worker.host.clone(),
-                            }),
-                        });
+
+                // for chain (no shuffle) downstreams, we don't need to create hanging channels
+                if !ctx.no_shuffle_target_fragment_ids.contains(fragment_id) {
```

Comment:
How about asserting `upstream_worker_id == worker.id` here?

Reply:
This index is created at the beginning based on the status of the existing actor, so for the chain fragment's upstream will not be found in the follow `actor_id_to_parallel_unit`